modules/web/js/langserver/langserver-channel.js

```diff
@@ -31,6 +31,12 @@ class LangserverChannel extends EventChannel {
         if (_.isNil(args.endpoint)) {
             throw 'Invalid Endpoint';
         }
+        // assign provided ws close handler.
+        if (!_.isNil(args.clientController.options)){
+            if (!_.isNil(args.clientController.options.wsCloseEventHandler)) {
```

Comment:
shall we use `if  (args.clientController.options && args.clientController.options.wsCloseEventHandler) {` here?